### PR TITLE
139 Make Vagrant Windows E2E tests use TLS connections to Conjur

### PR DESCRIPTION
**Pull request by sgnn7**
### What does this PR do?
- This change makes Vagrant tests use TLS connection back to Conjur (rather than a non-TLS one)

### What ticket does this PR close?
Connected to #139 

#### Change log
- [ ] The CHANGELOG has been updated, or
- [x] This PR does not include user-facing changes and doesn't require a CHANGELOG update

#### Test coverage
- [ ] This PR includes new unit and integration tests to go with the code changes, or
- [x] The changes in this PR do not require tests

#### Documentation
- [x] Docs (e.g. `README`s) were updated in this PR, and/or there is a follow-on issue to update docs, or
- [ ] This PR does not require updating any documentation

### CODE REVIEW

File: examples/puppetmaster/vagrant/windows2012/Vagrantfile [hunk 1]
```diff
@@ -21,6 +21,8 @@ Vagrant.configure("2") do |config|
     # Set VirtualBox VM name and set shared clipboard mode to bidirectional
     config.vm.provider :virtualbox do |vb|
         vb.name = "Windows2012"
+        vb.cpus = 3
```
Comment (diverdane) [inline examples/puppetmaster/vagrant/windows2012/Vagrantfile:24]: Nice. I forgot to tune the cpus/memory for this.

File: examples/puppetmaster/vagrant/README.md [hunk 2]
```diff
@@ -20,6 +20,7 @@
   * [Install Puppet Agent on the Windows VM](#install-puppet-agent-on-the-windows-vm)
   * [Clear Previous Provisioning Results (if Necessary)](#clear-previous-provisioning-results--if-necessary-)
```
Comment (diverdane) [inline examples/puppetmaster/vagrant/README.md:21]: This line (`Clear Previous Provisioning Results`) can be deleted from Table of Contents.

File: examples/puppetmaster/vagrant/3_run_puppet_agent.sh [hunk 3]
```diff
@@ -11,24 +11,40 @@ hostname="$VAGRANT_CWD"
...
+
 echo "Adding Conjur connection info to Windows registry"
 vagrant powershell -e -c "/vagrant/add_conjur_registry.ps1 $(conjur_host_port)"
 
 echo "Rotating Conjur API key for host 'node01'"
-conjur_ctl_container=$(docker ps | awk '/cyberark\/conjur-cli/{print $1}')
-node_api_key="$(docker exec $conjur_ctl_container conjur host rotate_api_key -h node01)"
+node_api_key="$(docker exec $conjur_cli_container conjur host rotate_api_key -h node01)"
+
+echo "Copying Puppet CA certs from master to host..."
```
[the remark this answers is not in the document]
Comment (sgnn7) [reply examples/puppetmaster/vagrant/3_run_puppet_agent.sh:27]: I fixed this a bit by adding support for server-side config. Let me know if it's not enough of a change.

File: examples/puppetmaster/vagrant/add_conjur_registry.ps1 [hunk 4]
```diff
@@ -20,6 +20,10 @@ if (-not($conjurHostPort)) {
     exit 1
 }
 
-reg ADD HKLM\Software\CyberArk\Conjur /f /v ApplianceUrl /t REG_SZ /d http://conjur:$conjurHostPort
+reg ADD HKLM\Software\CyberArk\Conjur /f /v ApplianceUrl /t REG_SZ /d https://conjur-https:$conjurHostPort
 reg ADD HKLM\Software\CyberArk\Conjur /f /v Version /t REG_DWORD /d 5
 reg ADD HKLM\Software\CyberArk\Conjur /f /v Account /t REG_SZ /d cucumber
+
+reg ADD HKLM\Software\CyberArk\Conjur /f /v CertFile /t REG_SZ /d "/vagrant/.tmp/conjur_ca.pem"
+# $ssh_certificate = Get-Content -Raw -Path /vagrant/.tmp/conjur_ca.pem
```
[the remark this answers is not in the document]
Comment (sgnn7) [reply examples/puppetmaster/vagrant/add_conjur_registry.ps1:28]: Oops - that should be "SS**L**" certificate - sill fix

File: examples/puppetmaster/vagrant/run_puppet_agent.ps1 [hunk 5]
```diff
@@ -22,10 +25,17 @@ if (-not($puppetSvrHostPort)) {
...
 }
 
+$vagrant_cert_dir = "/vagrant/.tmp"
+$puppet_cert_dir = "/ProgramData/PuppetLabs/puppet/etc/ssl/certs"
 
-# Clean any previous SSL certificates that were generated by the
-# Puppet master
+"Recreating Puppet SSL directory..."
 rm -r /ProgramData/PuppetLabs/puppet/etc/ssl
+New-Item "$puppet_cert_dir" `
```
[the remark this answers is not in the document]
Comment (sgnn7) [reply examples/puppetmaster/vagrant/run_puppet_agent.ps1:33]: Yeah, I fully agree - it's pretty weird lol

File: examples/puppetmaster/vagrant/run_puppet_agent.ps1 [hunk 6]
```diff
@@ -22,10 +25,17 @@ if (-not($puppetSvrHostPort)) {
     exit 1
 }
 
+$vagrant_cert_dir = "/vagrant/.tmp"
+$puppet_cert_dir = "/ProgramData/PuppetLabs/puppet/etc/ssl/certs"
 
-# Clean any previous SSL certificates that were generated by the
-# Puppet master
+"Recreating Puppet SSL directory..."
 rm -r /ProgramData/PuppetLabs/puppet/etc/ssl
```
[the remark this answers is not in the document]
Comment (sgnn7) [reply examples/puppetmaster/vagrant/run_puppet_agent.ps1:32]: Done

**Pull request merged**